src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DefaultProjectSnapshotManager.cs

```diff
@@ -30,6 +30,7 @@ internal class DefaultProjectSnapshotManager : ProjectSnapshotManagerBase
     // created lazily.
     private readonly Dictionary<string, Entry> _projects;
     private readonly HashSet<string> _openDocuments;
+    private readonly LoadTextOptions LoadTextOptions = new LoadTextOptions(SourceHashAlgorithm.Sha1);
```

Comment:
This was chosen to match https://github.com/dotnet/roslyn/commit/157d34acf5c19fc6780ac00ca09e58d40873864a#diff-cc1e8b8a45730c3fac9f64e9958ebf7a11230c3f6645f3f91709e85264688436R460 (and because it causes a ton of silent errors if you select `SourceHashAlgorithm.None`).

Reply:
We really shouldn't be using SHA1 for new code unless there is a reason to do it for backwards compatibility. Presumably there is in Roslyn, because thats public API, but we don't have that issue. If SHA256 works we should use it.

Reply:
Agree with @davidwengier. As a rule of thumb, I'd never personally pick a SHA without checking with @tmat first. :smile: